tests/unittests/test_sync.py

```diff
@@ -0,0 +1,138 @@
+            get_stream_catalog("companies", parent),
+            get_stream_catalog("conversations", child),
+            get_stream_catalog("tickets", parent),
+            get_stream_catalog("time_entries", child),
+            get_stream_catalog("groups", parent),
+        ]
+    }
+
+
+
+class TestSyncFunctions(unittest.TestCase):
+    """
+    Test `sync` function.
+    """
+
+    # NOTE: For `tickets` stream `sync_obj` is called 3 times
+    @parameterized.expand([
+        ["only_parent_selected", get_catalog(parent=True), ["companies", "tickets", "groups"], 5],
+        ["only_child_selected", get_catalog(child=True), ["conversations", "time_entries"], 3],
+        ["both_selected", get_catalog(parent=True, child=True), ["companies", "tickets", "groups", "conversations", "time_entries"], 5],
+        ["No_streams_selected", get_catalog(), [], 0],
+    ])
+    @mock.patch("singer.write_state")
+    @mock.patch("singer.write_schema")
+    @mock.patch("tap_freshdesk.streams.Stream.sync_obj")
+    def test_sync(self, name, mock_catalog, selected_streams, synced_streams, mock_sync_endpoint, mock_write_schemas, mock_write_state):
+        """
+        Test sync function.
+        """
+        client = mock.Mock()
+        sync(client, {'start_date': ""}, {}, mock_catalog)
+
+        # Verify write schema is called for selected streams
+        self.assertEqual(mock_write_schemas.call_count, len(selected_streams))
+        for stream in selected_streams:
+            mock_write_schemas.assert_any_call(stream, mock.ANY, mock.ANY)
+
+        # Verify sync object was called for syncing parent streams
+        self.assertEqual(mock_sync_endpoint.call_count, synced_streams)
+
+
+class TestWriteSchemas(unittest.TestCase):
+    """
+    Test `write_schemas` function.
+    """
+
+    mock_catalog = {"streams": [
+        get_stream_catalog("tickets"),
+        get_stream_catalog("time_entries"),
+        get_stream_catalog("conversations")
+    ]}
+
+    @parameterized.expand([
+        ["parents_selected", ["tickets"]],
+        ["child_selected", ["time_entries"]],
+        ["parent_and_child_selected", ["tickets", "conversations"]],
+    ])
+    @mock.patch("singer.write_schema")
+    def test_write_schema(self, name, selected_streams, mock_write_schema):
+        """
+        Test that only schema is written for only selected streams.
+        """
+        write_schemas("tickets", self.mock_catalog, selected_streams)
+        for stream in selected_streams:
+            # Verify write_schema function is called.
+            mock_write_schema.assert_any_call(stream, mock.ANY, mock.ANY)
+
+
+class TestGetStreamsToSync(unittest.TestCase):
+    """
+    Testcase for `get_stream_to_sync` in sync.
+    """
+
+    @parameterized.expand([
+        ['test_parent_selected', ["tickets"], ["tickets"]],
+        ['test_child_selected', ["conversations", "satisfaction_ratings"], ["conversations", "satisfaction_ratings", "tickets"]],
```

Comment:
```suggestion
        ['test_child_selected', ["conversations", "satisfaction_ratings"], ["tickets"]],
```

Reply:
Streams to sync should return parent streams only.

Reply:
Updated the code as well as the unittest for the streams_to_sync to return parent streams only